Dockerfile

```
@@ -0,0 +1,22 @@
+FROM alpine:3.13
+
+WORKDIR /git-server/
+
+RUN apk add --no-cache openssh git
+
+RUN ssh-keygen -A 
+
+COPY sshd_config /etc/ssh/sshd_config
+
+COPY start.sh /git-server/
+
+RUN git config --global init.defaultBranch main
+
+RUN mkdir /git-server/keys \
+  && adduser -D git \
```

Use the restricted git-shell:
```suggestion
  && adduser -s /usr/bin/git-shell -D git \
```